presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -335,6 +335,19 @@ public PlanWithProperties visitAggregation(AggregationNode node, StreamPreferred
                 // !isPresent() indicates the property was satisfied completely
                 preGroupedSymbols = groupingKeys;
             }
+            else {
+                // Enable segmented aggregation if the prefix of the sorted-by columns is a subset of the group by column.
+                // Use the longest prefix of the sorted-by columns that satisfy the requirement to maximize the segment aggregation's preGroupedSymbols
+                for (int i = child.getProperties().getLocalProperties().size(); i > 0; i--) {
+                    List<LocalProperty<VariableReferenceExpression>> sortedByKeysPrefix = child.getProperties().getLocalProperties().subList(0, i);
+                    List<Optional<LocalProperty<VariableReferenceExpression>>> matchResult = LocalProperties.match(sortedByKeysPrefix, LocalProperties.grouped(groupingKeys));
```

Comment:
This may not work if groupingKeys has a different order than the sortedByKeys. We may need to rearrange the order of the fields in groupingKeys according to the order in sortedByKeysPrefix. We can add a TODO to support this case later.